pkg/scalers/kafka_scaler.go

```diff
@@ -430,17 +462,29 @@ func (s *kafkaScaler) getTopicOffsets(partitions []int32) (map[int32]int64, erro
 		request.AddBlock(s.metadata.topic, partitionID, sarama.OffsetNewest, 1)
 	}
 
-	offsets := make(map[int32]int64)
-
 	// Step 2: send requests, one per broker, and collect offsets
+	resultCh := make(chan brokerOffsetResult, len(requests))
+	var wg sync.WaitGroup
+	wg.Add(len(requests))
 	for broker, request := range requests {
-		response, err := broker.GetAvailableOffsets(request)
+		go func(brCopy *sarama.Broker, reqCopy *sarama.OffsetRequest) {
```

Comment:
Do you remember why you needed to create copies here, opposed to your initial commit?

https://github.com/VerstraeteBert/keda/commit/e6073130304758ea5f87abf3f72f4f8bccf815eb#diff-03cf62956adcec86c0717723ab146472f0979221b0e5942a6e5460fa6d5e4f08R445

Reply:
A race condition: https://stackoverflow.com/questions/40326723/go-vet-range-variable-captured-by-func-literal-when-using-go-routine-inside-of-f